src/canisters/frontend/ic-certified-assets/src/lib.rs

```diff
@@ -24,15 +29,55 @@ thread_local! {
 
 #[update]
 #[candid_method(update)]
-fn authorize(other: Principal) {
+async fn authorize(other: Principal) {
+    let caller = caller();
+    let is_authorized = STATE.with(|s| s.borrow().is_authorized(&caller));
+    if is_authorized {
+        STATE.with(|s| {
+            if let Err(msg) = s.borrow_mut().authorize(&caller, other) {
```

Comment:
Since we've already checked `is_authorized` by this point, let's call `authorize_unconditionally` here and delete the `State::authorize` method.
```suggestion
            if let Err(msg) = s.borrow_mut().authorize_unconditionally(&caller, other) {
```